tmail-backend/jmap/extensions-api/src/main/scala/com/linagora/tmail/james/jmap/calendar/CalendarEventModifier.scala

```diff
@@ -0,0 +1,117 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.james.jmap.calendar
+
+import java.time.temporal.Temporal
+import java.time.{Instant, ZoneId, ZonedDateTime}
+import java.util
+import java.util.function.Consumer
+
+import com.google.common.base.Preconditions.{checkArgument => require}
+import com.google.common.collect.ImmutableList
+import com.linagora.tmail.james.jmap.model.{CalendarEventParsed, CalendarTimeZoneField, VEventTemporalUtil}
+import net.fortuna.ical4j.model.component.VEvent
+import net.fortuna.ical4j.model.property.{DtEnd, Sequence}
+import net.fortuna.ical4j.model.{Calendar, Component, Property, PropertyList}
+import net.fortuna.ical4j.validate.ValidationResult
+
+import scala.jdk.CollectionConverters._
+
+object CalendarEventModifier {
+  val MODIFIED_SEQUENCE_DEFAULT: Sequence = new Sequence("1")
+
+  case class NoUpdateRequiredException(message: String) extends RuntimeException(message)
+
+  def modifyEventTiming(originalCalendar: Calendar,
+                        newStartDate: ZonedDateTime,
+                        newEndDate: ZonedDateTime,
+                        validator: Consumer[VEvent] = (_: VEvent) => {}): Calendar = {
+    validateBeforeUpdate(originalCalendar, newStartDate, newEndDate)
+
+    val newCalendar = originalCalendar.copy()
+    val vEvent = newCalendar.getComponent[VEvent](Component.VEVENT).get()
+    validator.accept(vEvent)
+    updateEvent(vEvent, newStartDate, newEndDate)
```

Comment:
Please confirm me, double checking RFC if needed, that counter only can update start and end and no other property of the event.

Reply:
I have check RFC 5546 #section-3.2.7 and related examples,
but the RFC does not exactly list fields should be updated from COUNTER.

I have gathered scattered pieces of information mentioned in the RFC:
- DTSTART, DTEND (implemented)
- LOCATION, ATTENDEE (not yet implemented in this PR)

I checked the linagora openpaas, we right now only support propose (DTSTART, DTEND)
![image](https://github.com/user-attachments/assets/9a55393e-d734-4ee9-b6d7-ba666b2d14ad)

Google Calendar is same
![image](https://github.com/user-attachments/assets/6e74033c-78a1-4822-8c32-e4ac4798e95a)

Which choice should it be?

Reply:
We can likely start with start+end but we may create an enhancement, (timeboxed) to support updates on other fields as well.